providers/cluster.rb

```diff
@@ -0,0 +1,258 @@
+#
+# Cookbook Name:: rabbitmq
+# Provider:: cluster
+#
+# Author: Sunggun Yu <sunggun.dev@gmail.com>
+# Copyright (C) 2015 Sunggun Yu
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include Chef::Mixin::ShellOut
+
+use_inline_resources
+
+# Get ShellOut
+def get_shellout(cmd)
+  sh_cmd = Mixlib::ShellOut.new(cmd)
+  sh_cmd.environment['HOME'] = ENV.fetch('HOME', '/root')
+  sh_cmd
+end
+
+# Execute rabbitmqctl command with args
+def run_rabbitmqctl(*args)
+  cmd = "rabbitmqctl #{args.join(' ')}"
+  Chef::Log.debug("[rabbitmq_cluster] Executing #{cmd}")
+  cmd = get_shellout(cmd)
+  cmd.run_command
+  begin
+    cmd.error!
+    Chef::Log.debug("[rabbitmq_cluster] #{cmd.stdout}")
+  rescue
+    Chef::Application.fatal!("[rabbitmq_cluster] #{cmd.stderr}")
+  end
+end
+
+# Get cluster status result
+def cluster_status
+  # execute > rabbitmqctl cluster_status | sed "1d" | tr "\n" " " | tr -d " "
+  # rabbitmqctl cluster_status returns "Cluster status of node rabbit@rabbit1 ..." at the first line.
+  # To parse the result string, it is removed by sed "1d"
+  cmd = 'rabbitmqctl cluster_status | sed "1d" | tr "\n" " " | tr -d " "'
```

Comment:
@jjasghar @cmluciano 

It would be changed like this.
```
def cluster_status
  # execute > rabbitmqctl cluster_status"
  # To parse the result string, this function normalize the output string
  # - Removing first line
  # - Removing "... Done" : old version returns this
  cmd = 'rabbitmqctl cluster_status'
  Chef::Log.debug("[rabbitmq_cluster] Executing #{cmd}")
  cmd = get_shellout(cmd)
  cmd.run_command
  cmd.error!
  result = cmd.stdout.split(/\n/, 2).last.squeeze(' ').gsub(/\n/, '').gsub('...done.', '')
  Chef::Log.debug("[rabbitmq_cluster] rabbitmqctl cluster_status : #{result}")
  result
end
```